File: sdk/player.lua
bot = require('bot')

math.randomseed(os.time())

local s = bot.recv()
bot.log('Hello from the Moon! Side #' .. s .. '\n')

local used = {}
while true do
    local r, c = bot.recv():match('(%S+) (%S+)')
    local id = tonumber(r) * 3 + tonumber(c) + 1
    used[id] = true

    local sel
    repeat sel = math.random(9) until not used[sel]
    used[sel] = true

    bot.send(string.format('%d %d', math.floor((sel - 1) / 3), (sel - 1) % 3))
end

File: sdk/player.py
import bot
import random

bot.log('Hello from the snake! Side #' + bot.recv() + '\n')
random.seed()

used = [[False] * 3 for _ in range(3)]
while True:
    r, c = map(int, bot.recv().split())
    if r != -1: used[r][c] = True

    while True:
        r = random.randrange(0, 3)
        c = random.randrange(0, 3)
        if not used[r][c]: break

    used[r][c] = True
    bot.send('%d %d' % (r, c))
